### ISSUE TITLE
#[rustfmt::skip::macros(...)] doesn't work in impl blocks and trait defenitions

### ISSUE THREAD

**Issue by UltraMachine** [rust-lang/rustfmt#4167]
```rust
// Skipping fails in these 3 examples
impl Type {
    #[rustfmt::skip::macros(macros)]
    fn example() {
        macros!(
            "do not format"
        );
    }
}
impl Trait for Type {
    #[rustfmt::skip::macros(macros)]
    fn example() {
        macros!(
            "do not format"
        );
    }
}
trait Trait {
    #[rustfmt::skip::macros(macros)]
    fn example() {
        macros!(
            "do not format"
        );
    }
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=483272a09b3cbab362f9767c8769f045) with more examples

#[rustfmt::skip::attributes(...)] fails the same way

**Comment by Effnote** [comment]
Has this regressed? I'm easily able to reproduce this using the given playground link, and in my own code.

**Comment by ytmimi** [comment]
@Effnote the PR that originally addressed this was merged into what is now the `rustfmt-2.0.0-rc.2` branch. It was an experimental version that is unlikely to be released, but the team has tried to pull some changes from `rustfmt-2.0.0-rc.2` back into a release branch.

If you're interested, you could try porting #4209 into the codebase, though since this was released rustfmt added the [`skip_macro_invocations`](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#skip_macro_invocations) config option, and maybe that would work for your use case if you're using nightly rustfmt.
